integrations/Criteo/browser.js

```diff
@@ -0,0 +1,222 @@
+          navigator.userAgent
+        )
+      ? "m"
+      : "d";
+    this.fieldMapping = config.fieldMapping;
+    this.OPERATOR_LIST = ["eq", "gt", "lt", "ge", "le", "in"];
+  }
+
+  init() {
+    logger.debug("===in init Criteo===");
+    if (!this.accountId) {
+      return;
+    }
+    window.criteo_q = window.criteo_q || [];
+
+    ScriptLoader(
+      "Criteo",
+      `//dynamic.criteo.com/js/ld/ld.js?a=${this.accountId}`
+    );
+    window.criteo_q.push({ event: "setAccount", account: this.accountId });
+    window.criteo_q.push({ event: "setSiteType", type: this.deviceType });
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  isLoaded() {
+    logger.debug("in Criteo isLoaded");
+    return !!(window.criteo_q && window.criteo_q.push !== Array.prototype.push);
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  isReady() {
+    logger.debug("in Criteo isReady");
+    return !!(window.criteo_q && window.criteo_q.push !== Array.prototype.push);
+  }
+
+  handleCommonFields(rudderElement) {
+    const { message } = rudderElement;
+    const { properties } = message;
+
+    const setEmail = {};
+    const setZipcode = {};
+
+    const finalRequest = [
+      { event: "setCustomerId", id: md5(message.userId) },
+      { event: "setRetailerVisitorId", id: md5(message.anonymousId) },
+    ];
+
+    if (properties.email) {
+      const email = properties.email.trim().toLowerCase();
+      setEmail.event = "setEmail";
+      setEmail.hash_method = this.hashMethod;
+      setEmail.email = this.hashMethod === "md5" ? md5(email) : email;
+      finalRequest.push(setEmail);
+    }
+
+    if (properties.zipCode) {
+      setZipcode.event = "setZipcode";
+      setZipcode.zipCode = properties.zipCode || properties.zip;
+      finalRequest.push(setZipcode);
+    }
+
+    return finalRequest;
+  }
+
+  page(rudderElement) {
+    const { name, properties } = rudderElement.message;
+
+    const finalPayload = this.handleCommonFields(rudderElement);
+
+    if (
+      name === "home" ||
+      (this.url && this.url === window.location.href) ||
+      properties.url === this.url
+    ) {
+      const homeEvent = {
+        event: "viewHome",
+      };
+      finalPayload.push(homeEvent);
+    }
+
+    window.criteo_q.push(finalPayload);
+  }
+
+  track(rudderElement) {
+    const { event, properties } = rudderElement.message;
+
+    const finalPayload = this.handleCommonFields(rudderElement);
+
+    // Product tag
+    if (event === "Product Viewed") {
+      const viewItemObject = {
+        event: "viewItem",
+        item: String(properties.product_id),
+      };
+
+      if (properties.price && !Number.isNaN(parseFloat(properties.price))) {
+        viewItemObject.price = parseFloat(properties.price);
+      }
+
+      if (properties.availability === 1 || properties.availability === 0) {
+        viewItemObject.availability = properties.availability;
+      }
+      if (!viewItemObject.item) {
```

Comment:
let's add this check beforehand before even creating the `viewItemObject`

Reply:
handled